common/common.go

```diff
@@ -400,8 +406,74 @@ func containerEngineBuildV20180708(verbose bool, fpath string, ff *FuncFileV2018
 	return nil
 }
 
+func buildXDockerCommand(imageName, dockerfile string, buildArgs []string, noCache bool, architectures []string) []string {
+	var buildCommand = "buildx"
+	var name = imageName
+
+	args := []string{
+		buildCommand,
+		"build",
+		"-t", name,
+		"-f", dockerfile,
+		"--platform", strings.Join(architectures, ","),
+	}
+
+	if noCache {
+		args = append(args, "--no-cache")
+	}
+
+	if len(buildArgs) > 0 {
+		for _, buildArg := range buildArgs {
+			args = append(args, "--build-arg", buildArg)
+		}
+	}
+
+	if len(architectures) > 0 {
+		var arg = "ARCH=" + strings.Join(architectures, ",")
+		var label = "imageName=" + imageName
+		args = append(args, "--build-arg", arg)
+		args = append(args, "--label", label)
+		args = append(args, "--push")
+	}
+
+	args = append(args,
+		"--build-arg", "HTTP_PROXY",
+		"--build-arg", "HTTPS_PROXY",
+		".")
+
+	// Container engine type would be optional here
+	return args
+}
+
+func buildDockerCommand(imageName, dockerfile string, buildArgs []string, noCache bool) []string {
+	var name = imageName
+	args := []string{
+		"build",
+		"-t", name,
+		"-f", dockerfile,
+	}
+
+	if noCache {
```

Comment:
Code snippets of no-cache,build-arg,http-proxy are duplicated across DockerBuildx and DockerBuild methods. Can you refactor them?